### PR TITLE
feat: add 3D scan option to service request

### PR DESCRIPTION
**Pull request by Alchez**
**Ref:** [TASK-2019-00858](https://digithinkit.global/desk#Form/Task/TASK-2019-00858)

Requires https://github.com/DigiThinkIT/jhaudio_customizations/pull/1239.

<hr>

**Changelog:**
- [ ] Add a "3D Scan" option to the fit form
  - [ ] Show a banner indicating the new "3D Scan" option ([website configuration](https://jhaudio.com/desk#Form/Fit%20Correction%20Settings/Fit%20Correction%20Settings))
  - [ ] Let users know that the 3D scan is an additional charge ([website configuration](https://jhaudio.com/desk#List/Repair%20Issue%20Option/List))
  - [x] Add an "Item Group" link to Repair Issue Option to filter 3D scan for CIEMS
- [x] Capture the "3D scan" option from both,
  - [x] the Fit form, and
  - [x] any manual entries from the backend
- [x] Set a service item for the scan in the service request
  - [x] Add an "Is 3D Scan" check to identify issue on the Service Request
  - [x] If 3D scan is not possible (due to broken shells or shell wear), allow service reps to reject 3D scans while recording testing results
- [x] Carry forward the service item for IEM scan onto sales documents

<hr>

**Changes:**
- Added a "Default 3D Scan Service Item" to `Repair Settings` (with filters for service items)
- Added 3D scan options to the "Test Item" dialog
- If a default scan item is set, the system pulls that item into the Service Request after testing is completed

### CODE REVIEW

File: repairs/repairs/doctype/repair_settings/repair_settings.js [hunk 1]
```diff
@@ -2,13 +2,19 @@
 // For license information, please see license.txt
 
 frappe.ui.form.on('Repair Settings', {
-	refresh: function(frm) {
+	setup: (frm) => {
+		frm.set_query('default_3d_scan_item', () => {
+			return { filters: { "item_group": "Services" } };
+		})
```
Comment (vjFaLk) [inline repairs/repairs/doctype/repair_settings/repair_settings.js]: Codacy Issue found: [Missing semicolon. (semi)](https://app.codacy.com/app/DigiThinkIT/repairs/pullRequest?prid=4722558)

File: repairs/repairs/doctype/repair_settings/repair_settings.js [hunk 2]
```diff
@@ -2,13 +2,19 @@
 // For license information, please see license.txt
 
 frappe.ui.form.on('Repair Settings', {
-	refresh: function(frm) {
+	setup: (frm) => {
+		frm.set_query('default_3d_scan_item', () => {
```
Comment (vjFaLk) [inline repairs/repairs/doctype/repair_settings/repair_settings.js]: Codacy Issue found: [Strings must use doublequote. (quotes)](https://app.codacy.com/app/DigiThinkIT/repairs/pullRequest?prid=4722558)

File: repairs/public/js/warranty_claim.js [hunk 3]
```diff
@@ -73,7 +73,21 @@ frappe.ui.form.on("Warranty Claim", {
 
 			// Start testing the item
 			if (frm.doc.status == "To Test") {
-				var repair_btn = frm.add_custom_button(__("Test Item"), () => {
+				let requires_scan = {
+					left_ear: false,
+					right_ear: false
```
Comment (vjFaLk) [inline repairs/public/js/warranty_claim.js:78]: Codacy Issue found: [Identifier 'right_ear' is not in camel case. (camelcase)](https://app.codacy.com/app/DigiThinkIT/repairs/pullRequest?prid=4722558)

File: repairs/public/js/warranty_claim.js [hunk 4]
```diff
@@ -73,7 +73,21 @@ frappe.ui.form.on("Warranty Claim", {
 
 			// Start testing the item
 			if (frm.doc.status == "To Test") {
-				var repair_btn = frm.add_custom_button(__("Test Item"), () => {
+				let requires_scan = {
+					left_ear: false,
```
Comment (vjFaLk) [inline repairs/public/js/warranty_claim.js:77]: Codacy Issue found: [Identifier 'left_ear' is not in camel case. (camelcase)](https://app.codacy.com/app/DigiThinkIT/repairs/pullRequest?prid=4722558)

**Review by DeveloperMetal — APPROVED** [review]

**Pull request merged**